source/common/http/hash_policy.cc

```diff
@@ -39,15 +41,29 @@ class HeaderHashMethod : public HashMethodImplBase {
                                     const StreamInfo::FilterStateSharedPtr) const override {
     absl::optional<uint64_t> hash;
 
-    // TODO(mattklein123): Potentially hash on all headers.
     const auto header = headers.get(header_name_);
     if (!header.empty()) {
+      std::vector<std::string> rewrited_header_values;
```

Comment:
I would use `absl::InlinedVector` with a size of 1 here and below as it will almost always be one and/or just special case the single header case for performance reasons.

Reply:
got it